test/list_test.dart

```diff
@@ -361,4 +361,27 @@ Future<void> main([List<String>? args]) async {
     final result = team.players.query(r'name BEGINSWITH $0', ['J']);
     expect(result, [person]);
   });
+
+  test('List removeAt', () {
+    final config = Configuration.local([Team.schema, Person.schema]);
+    final realm = getRealm(config);
+
+    final alice = Person('Alice');
+    final bob = Person('Bob');
+    final carol = Person('Carol');
+    final dan = Person('Dan');
+    final team = Team('Class of 92', players: [alice, bob, carol, dan]);
+
+    realm.write(() => realm.add(team));
+    expect(team.players.length, 4);
+    expect(team.players, [alice, bob, carol, dan]);
+
+    expect(realm.write(() => team.players.removeAt(2)), carol);
+    //expect(team.players.length, 3);
+    expect(team.players, [alice, bob, dan]);
+
+    expect(realm.write(() => team.players.removeAt(0)), alice);
+    expect(team.players.length, 2);
+    expect(team.players, [bob, dan]);
+  });
```

Comment:
We need a check that removed items still exist in the ream, detached from the list. Similar to the tests for Clear https://github.com/realm/realm-dart/blob/77e730551b01a12c49c47367b16090a47942a557/test/list_test.dart#L98

Reply:
I have added checks for that as well